AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/CascadeDeleteOperation.swift

```diff
@@ -349,13 +369,15 @@ public class CascadeDeleteOperation<M: Model>: AsynchronousOperation {
         var savedDataStoreError: DataStoreError?
 
         guard !associatedModels.isEmpty else {
+            self.log.debug("[CascadeDelete.4] Begin syncing  \(models.count) \(modelName) model for deletion")
```

Comment:
If `associatedModels.isEmpty` is true then this log statement is performed. I still think this log line should be within the `syncDeletions(withModels:predicate:syncEngine:dataStoreError:completion:)` method because that way it will be logged in all code paths that call this method is called. The `syncDeleteions(withModels:predicate:associatedModels:)` method will sync the associated models first before syncing the parent models (as logged by this log line). If there are associated models, then this guard statement body is not executed, thus going further below to submit the associated models mutation events. Once `mutationEventsSubmitCompleted == associatedModels.count` on line 408 is true, then it will sync the parent models. With this log line being here, for the scenario where there are associated models, we are not logging the parent model's "begin syncing". If we put it inside `syncDeleteions(withModels:predicate:associatedModels:)` then any code paths calling this will log the number of parent models syncing count